### DISCUSSION TITLE
Initial conditions for multiple blocks

### DISCUSSION THREAD

**Discussion by ritamcj** [idaholab/moose#18422] [Q&A Modules: Phase field] [answered]
Hi Moose users,
I am looking into multiphase simulations in which each phase is defined by a block. I would like to assign initial values of a variable to each block, say c=0.6 for block 1 and c=0.4 for block 2. It would look something like:
```
[./eta1]
    order = FIRST
    family = LAGRANGE
    [./InitialCondition]
    variable = c
    type = ?
    value = '0.6   0.4'
    block = '1   2'
    [../]
  [../]
```
What could be the **type** in this code? I tried ConstantIC, ArrayConstantIC, ScalarComponentIC but could not get the syntax right. I would like to avoid Boundingbox as the blocks may not always be box-shaped as attached.

Warm regards,
Ritam Chatterjee
IIT Bombay, India

![Bicrystal](https://user-images.githubusercontent.com/81842276/126857428-16836c70-268d-4399-b4b8-514fcee7bef5.png)

**Comment by hugary1995** [comment, marked as answer]
You can follow this example: [https://github.com/idaholab/moose/blob/next/test/tests/ics/constant_ic/subdomain_constant_ic_test.i](https://github.com/idaholab/moose/blob/next/test/tests/ics/constant_ic/subdomain_constant_ic_test.i)

**Reply by ritamcj** [reply]
Thank you Gary. This works perfect!
Best regards,
Ritam
